### PR TITLE
[Tables] updating options

### PR DESCRIPTION
**Pull request by seankane-msft**
Updating options to make sure end user option struct is not manipulated 

- [ ] The purpose of this PR is explained in this or a referenced issue.
- [ ] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [ ] Tests are included and/or updated for code changes.
- [ ] Updates to [CHANGELOG.md][] are included.
- [ ] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go
[CHANGELOG.md]: https://github.com/Azure/azure-sdk-for-go/blob/main/CHANGELOG.md

### CODE REVIEW

File: sdk/data/aztables/go.mod [hunk 1]
```diff
@@ -4,8 +4,6 @@ go 1.16
 
 replace github.com/Azure/azure-sdk-for-go/sdk/azidentity => ../../azidentity
```
Comment (RickWinter) [inline sdk/data/aztables/go.mod:5]: This will fail your new check.

**Review by RickWinter — APPROVED** [review]

File: sdk/data/aztables/go.mod (hunk 1 above)
Comment (seankane-msft) [reply sdk/data/aztables/go.mod:5]: Yes, but it will be updated after the azidentity release and should pass

**Review by chlowell — APPROVED** [review]

**Pull request merged**